[EIP-3754] A vanilla non-fungible token standard (#3754)
ethereum/EIPs: EIPS/eip-3754.md

---
eip: 3754
title: A Vanilla Non-Fungible Token Standard
description: A vanilla NFT standard by removing data fields and functions related to `URI`.

> author: Simon Tian <simon@dtopia.me>

Reviewer: We prefer authors to use their github handle so our CI bot can tag them with notifications.
```suggestion
author: Simon Tian (@simontianx)
```

discussions-to: https://github.com/ethereum/EIPs/issues/3753
status: Draft
type: Standards Track,

> category (*only required for Standards Track): ERC

Reviewer: ```suggestion
category: ERC
```

created: 2021-08-21

> requires (*optional): <EIP number(s)>

Reviewer: ```suggestion
```

> requires (*optional): <EIP number(s)>

Reviewer: ```suggestion
```

replaces (*optional): <EIP number(s)>
---

## Abstract

> The ERC721 NFT standard has gained much popularity for its capability of being associated with external contents via the data field `URI`. However, this is also handicapping NFTs not having external associations. This standard is going to simplifiy ERC721 by removing data fields and functions related to `URI` and introduce a vanilla NFT that is ideal for representing abstract ownership such as rights. This can open up more possibilities by allowing layers of abstraction being built on top of it.

Reviewer: ```suggestion
The [ERC-721](./eip-721.md) NFT standard has gained much popularity for its capability of being associated with external contents via the data field `URI`. However, this is also handicapping NFTs not having external associations. This standard is going to simplifiy ERC-721 by removing data fields and functions related to `URI` and introduce a vanilla NFT that is ideal for representing abstract ownership such as rights. This can open up more possibilities by allowing layers of abstraction being built on top of it.
```

## Motivation
In developing `dtopia.me`, NFTs are used to represent abstract utilities. This is achieved by setting all `URI`s to empty strings. That means these NFTs are only functional on the `dtopia.me` platform. If such tokens are being listed on platforms such as `opensea`, `rarible` etc., a strange phenomenon can arise that no images or gifs would be shown in the front end, even though the tokens are completely ERC721 compliant. This introduces further difficulty in explaining what an NFT without `URI` can do to people who are deeply rooted in the current understanding of NFTs. A new standard for this type of NFTs must be developed to distinguish itself from others.

## Specification

> This standard is to remove data fields and functions related to `URI` from the current ERC721 implementation.

Reviewer: ```suggestion
This standard is to remove data fields and functions related to `URI` from the current ERC-721 implementation.
```

    function symbol() external view returns (string memory);
    function totalSupply() external view returns (uint256);
    function approve(address to, uint256 tokenId) external;
    function getApproved(uint256 tokenId) external view returns (address);
    function setApprovalForAll(address operator, bool approved) external;
    function isApprovedForAll(address owner, address operator) external view returns (bool);
    function transferFrom(address from, address to, uint256 tokenId) external;
    function safeTransferFrom(address from, address to, uint256 tokenId) external;
    function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory _data) external;
}
```
## Rationale
NFTs do not have to be associated with any external metadata or links to media contents. They can exist as standalone abstract atomic tokens. Such tokens allow layers of abstract utilities built on top of them. A strange phenomenon that tokens listed on NFT platforms do not have media contents to show should be avoided. New possibilities can be opened up because of this simplification. This can also reduce the costs in explaining this new type of NFTs.

## Backwards Compatibility

> ERC3754 is a simplified version of ERC721. There is no further backwards compatibility required.

Reviewer: ```suggestion
ERC-3754 is a simplified version of ERC-721. There is no further backwards compatibility required.
```

---
eip: 3754
title: A Vanilla Non-Fungible Token Standard
description: NFTs for representing abstract ownership
author: Simon Tian (@simontianx)
discussions-to: https://github.com/ethereum/EIPs/issues/3753
status: Draft

> type: Standards Track,

Reviewer: ```suggestion
type: Standards Track
```

category: ERC
created: 2021-08-21

> replaces (*optional): <EIP number(s)>

Reviewer: ```suggestion
```

---

## Abstract
In this standard, a non-fungible token stands as atomic existence and encourages
layers of abstraction built on top of it. Ideal for representing concepts like
rights, a form of abstract ownership. Such right can take the form of NFT options,
oracle membership, virtual coupons, etc., and can then be made liquid because of
this tokenization.

## Motivation
Non-fungible tokens are popularized by the [ERC-721](./eip-721.md) NFT standard
for representing "ownership over digital or physical assets". Over the course of
development, reputable NFT projects are about crypto-assets, digital collectibles,
etc. The proposed standard aims to single out a special type of NFTs that are
ideal for representing abstract ownership such as rights. Examples include the
right of making a function call to a smart contract, an NFT option that gives

> the owner the right but not obligation to purchase an ERC721 NFT, and the prepaid

Reviewer: ```suggestion
the owner the right but not obligation to purchase an ERC-721 NFT, and the prepaid
```
